pandas/tests/plotting/test_frame.py

```diff
@@ -1006,6 +1006,40 @@ def test_scatter_colors(self):
                                     np.array([1, 1, 1, 1], dtype=np.float64))
 
     @pytest.mark.slow
+    def test_plot_scatter_with_s(self):
+        data = np.array([[3.1, 4.2, 1.9],
```

Comment:
Could you add a test with some more variety in the data? I'm thinking

- all negative
- some negative some positive
- constant values
- some missing values
- some very large values

You don't need to do the whole plot for these tests, just verify that the values you get for `s` look correct. You may want to refactor you logic getting this to make it easier to test.

Reply:
OK. For these cases (negatives, NaN etc) I guess we want to have the same behaviour as what we'd get by passing s=df[s] directly ? That is:
* make the bubble plots only with points that have positive values (discard NaN and negative data points)
* throw a warning in case of negative values
?